store/src/java/com/zimbra/cs/service/formatter/MobileConfigFormatter.java

```diff
@@ -225,39 +241,48 @@ public void formatCallback(UserServletContext context) throws ServiceException {
                     "http://www.apple.com/DTDs/PropertyList-1.0.dtd");
             transformer.setOutputProperty(OutputKeys.DOCTYPE_PUBLIC, doctype.getPublicId());
             transformer.setOutputProperty(OutputKeys.DOCTYPE_SYSTEM, doctype.getSystemId());
-            FileBufferedWriter fileBufferedWriter = null;
-            try {
-                fileBufferedWriter = new FileBufferedWriter(
-                        context.resp.getWriter(),
-                        FileBufferedWriter.MAX_BUFFER_SIZE);
+            try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
                 DOMSource domSource = new DOMSource(document);
-                StringWriter writer = new StringWriter();
-                StreamResult responseStream = new StreamResult(writer);
+                Result responseStream = new StreamResult(baos);
                 transformer.transform(domSource, responseStream);
-                ZimbraLog.misc.debug(writer.toString());
-                fileBufferedWriter.write(writer.toString());
+                byte[] signedConfig = signConfig(server, baos.toByteArray());
```

Comment:
Can we check if private key is there on domain if not then use the server one?

Reply:
Done.